### DISCUSSION TITLE
Does spacy-cpp support Chinese?

### DISCUSSION THREAD

**Discussion by PythonCancer** [d99kris/spacy-cpp#21] [Q&A]
May I ask if your encapsulation of the space library also supports Chinese word segmentation, annotation, and entity naming functions?

**Comment by d99kris** [comment]
Hi @PythonCancer - I believe the Chinese support of `spacy-cpp` should match that of the underlying Python library `spacy`.

I'm not very well-versed in Chinese, so I cannot say whether it's working well. But here's an example:
```c++
#include <iostream>
#include <spacy/spacy>

int main()
{
  Spacy::Spacy spacy;
  auto nlp = spacy.load("zh_core_web_sm");
  auto doc = nlp.parse("这是一句话。");
  for (auto& token : doc.tokens())
    std::cout << token.text() << " [" << token.pos_() << "]\n";
}
```
The output is:
```
这是 [VERB]
一句话 [NOUN]
。 [PUNCT]
```
